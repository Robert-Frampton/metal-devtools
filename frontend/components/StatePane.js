import Component, {Config} from 'metal-jsx';
import {isPlainObject, keys} from 'lodash';

import FlashStateValue from './FlashStateValue';
import NodeName from './NodeName';

class StatePane extends Component {
	created() {
		this.inspectComponent = this.inspectComponent.bind(this);
	}

	inspectComponent() {
		const {component, onInspectDOM} = this.props;

		onInspectDOM(component.id);
	}

	render() {
		const {data = null, name} = this.props.component;

		const dataExists = isPlainObject(data);

		return (
			<div class="state-pane-container">
				<div class="header">
					{'Component: '}

					{name &&
						<span>
							<NodeName name={name} />

							<a class="see-in-dom" href="javascript:;" onClick={this.inspectComponent}>{'(Click to See Element)'}</a>
						</span>
					}
				</div>

				{dataExists &&
					keys(data).map(
						dataKey => {
							const stateObj = data[dataKey];

							return (
								<div class="category" key={`${name}-${dataKey}`}>
									<div class="name">{`${dataKey}:`}</div>

									<ul class="data">
										{
											keys(stateObj).map(
												stateObjKey => (
													<li key={`${name}-${dataKey}-${stateObjKey}`}>
														<b class="key">{`${stateObjKey}: `}</b>
														<FlashStateValue elementClasses="value" value={stateObj[stateObjKey]} />
													</li>
												)
											)
										}
									</ul>
								</div>
							);
						}
					)
				}

				{!dataExists &&
					<div>
						<i>{'No Component Data'}</i>
					</div>
				}
			</div>
		);
	}
}

StatePane.PROPS = {
	component: Config.any(),
	onInspectDOM: Config.func()
};

export default StatePane;
