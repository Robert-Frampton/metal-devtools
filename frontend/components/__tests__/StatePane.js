jest.unmock('../../lib/getComponentById');
jest.unmock('../../lib/processStateValues');
jest.unmock('../NodeName');
jest.unmock('../Resize');
jest.unmock('../StatePane');

import StatePane from '../StatePane';

describe('StatePane', () => {
	it('should render', () => {
		const component = new StatePane();

		expect(snap(component)).toMatchSnapshot();
	});
});
