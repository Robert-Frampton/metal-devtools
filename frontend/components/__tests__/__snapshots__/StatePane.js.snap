exports[`StatePane should render 1`] = `undefined`;
